src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrState = [];
+  let newState = { ...state };
+
+  actions.forEach((action) => {
+    if (action.type === 'clear') {
+      newState = {};
+      arrState.push({ ...newState });
+    }
+
+    if (action.type === 'removeProperties') {
+      action.keysToRemove.forEach((key) => delete newState[key]);
+      arrState.push({ ...newState });
+    }
+
+    if (action.type === 'addProperties') {
+      newState = {
+        ...newState,
+        ...action.extraData,
+      };
+      arrState.push({ ...newState });
+    }
```

Comment:
[CODE STYLE]: Use switch statement if you have limited amount of conditions. Instead of using multiple if statements, you should use a switch statement to handle the different action types. This will make your code cleaner and more maintainable.